File: server-dot-net/Model/LoanRequest.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server_dot_net.Model
{
    public class LoanRequest
    {
        [Key]
        public long Id { get; set; }

        public double Amount { get; set; }

        public string Reason { get; set; }

        public string Status { get; set; } // pending, approved, rejected

        public DateTime RequestDate { get; set; }

        [ForeignKey("Customer")]
        public long CustomerId { get; set; }

        public Customer Customer { get; set; }
    }
}

File: server-spring-boot/src/main/java/com/finflow/server/entity/LoanRequest.java
package com.finflow.server.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;


@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LoanRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Double amount;
    private String reason;
    private String status; // pending, approved, rejected
    private LocalDate requestDate;

    @ManyToOne
    private Customer customer;
}